main.cpp

```diff
@@ -127,17 +155,152 @@ struct IntType
             heapInt = nullptr;
         }
 
-        operator int() { return *heapInt; }
+        operator int() const { return *heapInt; }
 
         IntType& add( int operand );
         IntType& subtract( int operand );
         IntType& multiply( int operand );  
         IntType& divide( int operand );
+
+        IntType& pow( const FloatType& operand );
+        IntType& pow( const DoubleType& operand );
+        IntType& pow( const IntType& operand );
+        IntType& pow( int operand );
     
     private:
         int* heapInt = nullptr;
+        IntType& powInternal( int x );
 };
 
+//point constructor implementations
+Point::Point(FloatType& a, FloatType& b) : x(a), y(b) {}
+
+Point::Point(DoubleType& a, DoubleType& b) : x( static_cast<float>(a) ), y( static_cast<float>(b) ) {}
+
+Point::Point(IntType& a, IntType& b) : x( static_cast<float>(a) ), y( static_cast<float>(b) ) {}
+
+//point multiply UDT implementations
+Point& Point::multiply( FloatType& m )
+{
+    x *= m;
+    y *= m;
+    return *this;
+}
+
+Point& Point::multiply( DoubleType& m )
+{
+    x *= static_cast<float>(m);
+    y *= static_cast<float>(m);
+    return *this;
+}
+
+Point& Point::multiply( IntType& m )
+{
+    x *= static_cast<float>(m);
+    y *= static_cast<float>(m);
+    return *this;
+}
```

Comment:
all of these should call `multiply(float)` internally.  Look at what `Point::multiply(float)` returns.
- [ ] revise

Reply:
Fixed, but a question on line 186... without the static cast on the float type reference I get an infinite function warning, even though the float operator is implemented in the FloatType class. 
Is the static cast required so the compiler knows which version of the overloaded multiply function to call (the base one that accepts a float vs the UDT Float Type)?